.github/workflows/build_and_deploy.yml

```diff
@@ -59,36 +59,57 @@ jobs:
     runs-on: ubuntu-latest
     needs: [data-validation]
     steps:
+    - id: files
+      uses: jitterbit/get-changed-files@v1
+    - run: |
+          for changed_file in steps.files.outputs.all; do
+            echo " ${changed_file} changed "
+          done
     - uses: actions/checkout@v1
     - name: Set up Python 3.7
       uses: actions/setup-python@v1
       with:
         python-version: 3.7
     - name: Install dependencies
       run: |
-        python -m pip install --upgrade "pip<20"
-        pip install -r requirements-dev.txt
+          python -m pip install --upgrade "pip<20"
+          pip install -r requirements-dev.txt
     - name: Cross-validate NLU model
-      run: |
-        rasa test nlu -f 3 --cross-validation
-        python .github/workflows/format_results.py
+      if: |
+          contains(  steps.files.outputs.all, 'data/nlu/' ) 
+          || contains(  steps.files.outputs.all, 'config.yml' )
+      run: |  
+          rasa test nlu -f 3 --cross-validation
+          python .github/workflows/format_results.py
     - name: post cross-val results to PR
+      if: steps.cvnlu.outcome == 'success'
       uses: amn41/comment-on-pr@comment-file-contents
       continue-on-error: true
       env:
         GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
       with:
         msg: results.md
     - name: Train Model
+      if: |
+          contains(  steps.files.outputs.all, 'data/' ) 
+          || contains(  steps.files.outputs.all, 'config.yml' ) 
+          || contains(  steps.files.outputs.all, 'domain.yml' )
       working-directory: ${{ github.workspace }}
       run: |
         rasa train
     - name: Test Training Stories
+      if: |
+          ( contains(  steps.files.outputs.all, 'data/core/' ) 
+          || contains(  steps.files.outputs.all, 'config.yml' ) 
+          || contains(  steps.files.outputs.all, 'domain.yml' ) )
+          && steps.train.outcome == 'success'
       working-directory: ${{ github.workspace }}
       run: |
         rasa test core --stories test/test_stories.md --fail-on-prediction-errors
     - name: Upload model
-      if: github.ref == 'refs/heads/master'
+      if: |
+          github.ref == 'refs/heads/master'
+          && steps.train.outcome == 'success'
       uses: actions/upload-artifact@master
       with:
         name: model
```

Comment:
What do you think about action server images, should they only be built if the dockerfile or sth in the `actions/` folder changes?

Reply:
yeah, I think that makes sense. I'll add that condition